waitress/task.py

```diff
@@ -357,23 +353,10 @@ class ErrorTask(Task):
 
     def execute(self):
         e = self.request.error
-        body = '%s\r\n\r\n%s' % (e.reason, e.body)
-        tag = '\r\n\r\n(generated by waitress)'
-        body = body + tag
-        self.status = '%s %s' % (e.code, e.reason)
-        cl = len(body)
-        self.content_length = cl
-        self.response_headers.append(('Content-Length', str(cl)))
-        self.response_headers.append(('Content-Type', 'text/plain'))
-        if self.version == '1.1':
```

Comment:
Where are you handling this part in the new code? We do not need to send back `Connection: close` if requested by the client, and always append it if the client is HTTP 1.0.

Was this already being done elsewhere in the code and that is why it is longer required?

Reply:
I stared at this a while and from what I can tell since both `ErrorTask` and `WSGITask` subclass `Task` and `Task.write` invokes `Task.build_response_headers` and since that method seems to handle setting the connection header correctly based on version - it should not be necessary for `ErrorTask` to do it itself and the old code was redundant. I have not confirmed this and I appreciate that you noticed it. I figured I'd find out more when I went to write tests.

Reply:
:+1: I think you may be right.